From a6c2b3908073f8ea0c640c205278d017933e78b8 Mon Sep 17 00:00:00 2001
From: Lu Yahan <yahan@iscas.ac.cn>
Date: Wed, 4 Jan 2023 12:23:08 +0800
Subject: [PATCH] [riscv] Remove unnecessary unbound label count

The bind_to function doesn't link branch long to trampoline, so it doesn't need to add  unbound_labels_count_.

Change-Id: I2e3861a38eb65c285f19accb12bccb9f4c9fcfb1
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/4133426
Auto-Submit: Yahan Lu <yahan@iscas.ac.cn>
Reviewed-by: ji qiu <qiuji@iscas.ac.cn>
Commit-Queue: ji qiu <qiuji@iscas.ac.cn>
Cr-Commit-Position: refs/heads/main@{#85103}
---
 src/codegen/riscv/assembler-riscv.cc | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/codegen/riscv/assembler-riscv.cc b/src/codegen/riscv/assembler-riscv.cc
index 668e3fa9a81..c7773c57cd5 100644
--- a/src/codegen/riscv/assembler-riscv.cc
+++ b/src/codegen/riscv/assembler-riscv.cc
@@ -782,10 +782,6 @@ int32_t Assembler::branch_long_offset(Label* L) {
       L->link_to(pc_offset());
     } else {
       L->link_to(pc_offset());
-      if (!trampoline_emitted_) {
-        unbound_labels_count_++;
-        next_buffer_check_ -= kTrampolineSlotsSize;
-      }
       DEBUG_PRINTF("\tstarted link\n");
       return kEndOfJumpChain;
     }
-- 
2.35.1

